### PR TITLE
Fix/3881/header indentation stacks signing

### LINKED ISSUE #3881: Fix header indentation for Stacks transaction signing
For some reason the header text is now bumped to the right instead of aligned to the left:

Screenshot 2023-06-15 at 15 09 09

I believe I've also since an instance where it's entirely center aligned, though I can't recall how to recreate atm.

### PR DESCRIPTION
**Pull request by pete-watters**
> Try out this version of the Hiro Wallet - download [extension builds](https://github.com/hirosystems/wallet/actions/runs/5409919887).

This PR adds a fix to correctly align items in the signature header. 

This fixes the issue as you can see below, but I think we can further improve on it:

Kapture 2023-06-28 at 14 36 08

- You can see that the items move around based on the amounts in the account
- We use JS to truncate the display name
- We could probably refactor this to layout the header with a flexbox / CSS grid and make it less jumpy

I have made two small refactors also in separate commits which I can exclude from here if needed

### CODE REVIEW

File: src/app/pages/home/components/account-area.tsx [hunk 1]
```diff
@@ -4,14 +4,14 @@ import { Stack, StackProps } from '@stacks/ui';
...
 import { CurrentAccountAvatar } from '@app/features/current-account/current-account-avatar';
 import { CurrentAccountName } from '@app/features/current-account/current-account-name';
-import { useCurrentAccountNativeSegwitAddressIndexZero } from '@app/store/accounts/blockchain/bitcoin/native-segwit-account.hooks';
+import { useCurrentAccountNativeSegwitIndexZeroSigner } from '@app/store/accounts/blockchain/bitcoin/native-segwit-account.hooks';
 import { useCurrentStacksAccount } from '@app/store/accounts/blockchain/stacks/stacks-account.hooks';
 
 import { AccountTotalBalance } from '../../../components/account-total-balance';
 
 export const CurrentAccount = memo((props: StackProps) => {
   const currentAccount = useCurrentStacksAccount();
```
Comment (pete-watters) [inline src/app/pages/home/components/account-area.tsx]: I refactored this as I saw `useCurrentAccountNativeSegwitAddressIndexZero` is deprecated when I was reviewing the code

File: src/app/features/current-account/popup-header.tsx [hunk 2]
```diff
@@ -35,7 +35,7 @@ function PopupHeaderSuspense({ displayAddresssBalanceOf = 'stx' }: PopupHeaderPr
     <PopupHeaderLayout>
       <Flag align="middle" img={<CurrentAccountAvatar size="24px" fontSize="10px" />}>
         <SpaceBetween>
```
Comment (pete-watters) [inline src/app/features/current-account/popup-header.tsx:37]: This is the main fix to the alignment

File: src/app/features/current-account/current-account-name.tsx [hunk 3]
```diff
@@ -26,10 +26,10 @@ const AccountNameSuspense = memo((props: BoxProps) => {
   const currentAccount = useCurrentStacksAccount();
   const name = useCurrentAccountDisplayName();
   if (!currentAccount || typeof currentAccount.index === 'undefined') return null;
```
Comment (pete-watters) [inline src/app/features/current-account/current-account-name.tsx:28]: I added this comment so I can flag the code here. It seems a bit complicated - altering the name based on a char limit. 

We could do this using CSS and setting a max width of the account name

File: src/app/components/typography.tsx [hunk 4]
```diff
@@ -19,50 +19,42 @@ const openSauceMetrics = {
 const h1 = {
```
Comment (pete-watters) [inline src/app/components/typography.tsx:19]: I got rid of `leading` as it's not a valid CSS prop and I noticed it when investigating this. 

Screenshot 2023-06-28 at 09 55 22

`leading` and `line-height` are usually the same in design terms but if I swap to `lineHeight` here it really effects the UI so I think this is un-needed code. 

I may have missed something though so I can drop this commit if needed

File: src/app/pages/home/components/account-area.tsx (hunk 1 above)
Comment (pete-watters) [reply src/app/pages/home/components/account-area.tsx]: Reading #3940 and I'll probably need to account for that here too as bitcoin address is sometimes `undefined`

Comment (kyranjamie) [reply src/app/pages/home/components/account-area.tsx]: Let's revert this change as the behaviours are subtly different.

**Review by kyranjamie — APPROVED** [review]
LGTM besides signer change

File: src/app/pages/home/components/account-area.tsx (hunk 1 above)
Comment (pete-watters) [reply src/app/pages/home/components/account-area.tsx]: Thanks, I'll remove that commit

**Pull request merged**